### PR TITLE
HDDS-1329. Update documentation for Ozone-0.4.0 release. Contributed By Ajay Kumar.

### PR DESCRIPTION
**Pull request by ajayydv**

### CODE REVIEW

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+Renewer:    User responsible for renewing the token.
+Issue date:  Time at which token was issued.
+Max date:    Time after which token can’t be renewed. 
+
+Token operations like get, renew and cancel can only be performed over an Kerberos authenticated connection. Clients can use delegation token to establish connection with OzoneManager and perform any file system/object store related operations like, listing the objects in a bucket or creating a volume etc.
+
+#### Block Tokens ####
+Block tokens are similar to delegation tokens in sense that they are signed by OzoneManager. But this is where similarity between two stops. Block tokens are created by OM (OzoneManager) when a client request involves interaction with DataNodes. Unlike delegation tokens there is no client API to request block tokens. Instead they are handled transparently for client. Block tokens are embedded directly into client request responses. It means that clients don’t need to fetch it explicitly from Ozone Manager. This is handled implicitly inside ozone  client. Datanodes validates block tokens from clients for every client connection.
```
Comment (xiaoyuyao) [inline hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: Slightly edited version:
Block tokens are similar to delegation tokens in sense that they are signed by OzoneManager. Block tokens are created by OM (OzoneManager) when a client request involves interaction with DataNodes such as read/write Ozone keys. Unlike delegation tokens there is no client API to request block tokens. Instead, they are handed transparently to client along with key/block locations. Block tokens are validated by Datanodes when receiving read/write requests from clients. Block token can't be renewed explicitly by client. Client with expired block token will need to refetch the key/block locations to get new block tokens.

Comment (ajayydv) [reply hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: updated.

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+### Tokens ###
+Tokens are widely used in Hadoop to achieve lightweight authentication without compromising on security. Main motivation for using tokens inside Ozone is to prevent the unauthorized access while keeping the protocol lightweight and without sharing secret over the wire. Ozone utilizes three types of token:
+
+#### Delegation token ####
+Once client establishes their identity via kerberos they can request a delegation token from OzoneManager. This token can be used by a client to prove its identity until the token expires. Like Hadoop delegation tokens, an Ozone delegation token has 3 important fields:
+
+Renewer:    User responsible for renewing the token.
+Issue date:  Time at which token was issued.
+Max date:    Time after which token can’t be renewed.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: whitespace:end of line

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+#### S3Token ####
+Like block tokens S3Tokens are handled transparently for clients. It is signed by S3secret created by client. S3Gateway creates this token for every s3 client request. To create an S3Token user must have a S3 secret.
+
+### Certificates ###
+Apart from kerberos and tokens Ozone utilizes certificate based authentication for Ozone service components. To enable this, SCM (StorageContainerManager) bootstraps itself as an Certificate Authority when security is enabled. This allows all daemons inside Ozone to have an SCM signed certificate. Below is brief descriptions of steps involved:
+Datanodes and OzoneManagers submits a CSR (certificate signing request) to SCM.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: whitespace:end of line

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+#### S3Token ####
+Like block tokens S3Tokens are handled transparently for clients. It is signed by S3secret created by client. S3Gateway creates this token for every s3 client request. To create an S3Token user must have a S3 secret.
+
+### Certificates ###
+Apart from kerberos and tokens Ozone utilizes certificate based authentication for Ozone service components. To enable this, SCM (StorageContainerManager) bootstraps itself as an Certificate Authority when security is enabled. This allows all daemons inside Ozone to have an SCM signed certificate. Below is brief descriptions of steps involved:
+Datanodes and OzoneManagers submits a CSR (certificate signing request) to SCM. 
+SCM verifies identity of DN (Datanode) or OM via Kerberos and generates a certificate.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: whitespace:end of line

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+#### S3Token ####
+Like block tokens S3Tokens are handled transparently for clients. It is signed by S3secret created by client. S3Gateway creates this token for every s3 client request. To create an S3Token user must have a S3 secret.
+
+### Certificates ###
+Apart from kerberos and tokens Ozone utilizes certificate based authentication for Ozone service components. To enable this, SCM (StorageContainerManager) bootstraps itself as an Certificate Authority when security is enabled. This allows all daemons inside Ozone to have an SCM signed certificate. Below is brief descriptions of steps involved:
+Datanodes and OzoneManagers submits a CSR (certificate signing request) to SCM. 
+SCM verifies identity of DN (Datanode) or OM via Kerberos and generates a certificate. 
+This certificate is used by OM and DN to prove their identities.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: whitespace:end of line

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+Generating audit logs is only half the job, so Ozone also provides AuditParser - a sqllite based command line utility to parse/query audit logs with predefined templates(ex. Top 5 commands) and options for custom query. Once the log file has been loaded to AuditParser, one can simply run a template as shown below:
+ozone auditparser <path to db file> template top5cmds
+
+Similarly, users can also execute custom query using:
+ozone auditparser <path to db file> query "select * from audit where level=='FATAL'"
+
+## Transparent Data Encryption ##
+Ozone TDE borrows heavily from HDFS TDE. So setup process and usage is also very similar. When TDE is enabled an admin can create encryption keys and encryption zones. Any data written to encryption zone is encrypted.
+
+To know more about how to setup a secure Ozone cluster refer to [How to setup secure Ozone cluster]("SetupSecureOzone.md")
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+## Kerberos ##
+Configuration for service daemons:
+
+Property|Description
+--------|------------------------------------------------------------
+hdds.scm.kerberos.principal     | The SCM service principal. Ex scm/_HOST@REALM.COM_
+hdds.scm.kerberos.keytab.file   |The keytab file used by SCM daemon to login as its service principal.
+ozone.om.kerberos.principal     |The OzoneManager service principal. Ex om/_HOST@REALM.COM
+ozone.om.kerberos.keytab.file   |The keytab file used by SCM daemon to login as its service principal.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+hdds.scm.http.kerberos.principal|SCM http server service principal. 
+hdds.scm.http.kerberos.keytab   |The keytab file used by SCM http server to login as its service principal.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+## Tokens ##
+
+## Delegation token ##
+Delegation tokens are enabled by default when security is enabled.
+
+## Block Tokens ##
+hdds.block.token.enabled     | true
+-----------------------------|------
+
+## S3Token ##
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+S3 token are enabled by default when security is enabled. 
+To use S3 tokens users need to perform following steps:
+* S3 clients should get the secret access id and user secret from OzoneManager.
+```
+ozone s3 getsecret
+```
+* Setup secret in aws configs:
+```
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+aws configure set default.s3.signature_version s3v4
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+S3 token are enabled by default when security is enabled. 
+To use S3 tokens users need to perform following steps:
+* S3 clients should get the secret access id and user secret from OzoneManager.
+```
+ozone s3 getsecret
+```
+* Setup secret in aws configs:
+```
+aws configure set default.s3.signature_version s3v4  
+aws configure set aws_access_key_id ${accessId}
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+To use S3 tokens users need to perform following steps:
+* S3 clients should get the secret access id and user secret from OzoneManager.
+```
+ozone s3 getsecret
+```
+* Setup secret in aws configs:
+```
+aws configure set default.s3.signature_version s3v4  
+aws configure set aws_access_key_id ${accessId}  
+aws configure set aws_secret_access_key ${secret}
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+## Certificates ##
+Certificates are used internally inside Ozone. Its enabled be default when security is enabled.
+
+## Authorization ##
+Default access authorizer for Ozone approves every request. It is not suitable for production environments. It is recommended that clients use ranger plugin for Ozone to manage authorizations.
+
+ozone.acl.authorizer.class     | org.apache.ranger.authorization.ozone.authorizer.RangerOzoneAuthorizer
+-------------------------------|----------------------------------------------------
+
+## TDE ##
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 357 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1096 | trunk passed |
| +1 | mvnsite | 21 | trunk passed |
| +1 | shadedclient | 1778 | branch has no errors when building and testing our client artifacts. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 21 | the patch passed |
| +1 | mvnsite | 17 | the patch passed |
| -1 | whitespace | 0 | The patch has 13 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply |
| +1 | shadedclient | 743 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | asflicense | 29 | The patch does not generate ASF License warnings. |
| | | 3142 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-687/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/687 |
| Optional Tests |  dupname  asflicense  mvnsite  |
| uname | Linux 655b993edb99 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / be488b6 |
| maven | version: Apache Maven 3.3.9 |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-687/1/artifact/out/whitespace-eol.txt |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs U: hadoop-hdds/docs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-687/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+
+### Tokens ###
+Tokens are widely used in Hadoop to achieve lightweight authentication without compromising on security. Main motivation for using tokens inside Ozone is to prevent the unauthorized access while keeping the protocol lightweight and without sharing secret over the wire. Ozone utilizes three types of token:
+
+#### Delegation token ####
+Once client establishes their identity via kerberos they can request a delegation token from OzoneManager. This token can be used by a client to prove its identity until the token expires. Like Hadoop delegation tokens, an Ozone delegation token has 3 important fields:
+
+Renewer:    User responsible for renewing the token.
+Issue date:  Time at which token was issued.
+Max date:    Time after which token can’t be renewed.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: whitespace:end of line

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+#### S3Token ####
+Like block tokens S3Tokens are handled transparently for clients. It is signed by S3secret created by client. S3Gateway creates this token for every s3 client request. To create an S3Token user must have a S3 secret.
+
+### Certificates ###
+Apart from kerberos and tokens Ozone utilizes certificate based authentication for Ozone service components. To enable this, SCM (StorageContainerManager) bootstraps itself as an Certificate Authority when security is enabled. This allows all daemons inside Ozone to have an SCM signed certificate. Below is brief descriptions of steps involved:
+Datanodes and OzoneManagers submits a CSR (certificate signing request) to SCM.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: whitespace:end of line

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+#### S3Token ####
+Like block tokens S3Tokens are handled transparently for clients. It is signed by S3secret created by client. S3Gateway creates this token for every s3 client request. To create an S3Token user must have a S3 secret.
+
+### Certificates ###
+Apart from kerberos and tokens Ozone utilizes certificate based authentication for Ozone service components. To enable this, SCM (StorageContainerManager) bootstraps itself as an Certificate Authority when security is enabled. This allows all daemons inside Ozone to have an SCM signed certificate. Below is brief descriptions of steps involved:
+Datanodes and OzoneManagers submits a CSR (certificate signing request) to SCM. 
+SCM verifies identity of DN (Datanode) or OM via Kerberos and generates a certificate.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: whitespace:end of line

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+#### S3Token ####
+Like block tokens S3Tokens are handled transparently for clients. It is signed by S3secret created by client. S3Gateway creates this token for every s3 client request. To create an S3Token user must have a S3 secret.
+
+### Certificates ###
+Apart from kerberos and tokens Ozone utilizes certificate based authentication for Ozone service components. To enable this, SCM (StorageContainerManager) bootstraps itself as an Certificate Authority when security is enabled. This allows all daemons inside Ozone to have an SCM signed certificate. Below is brief descriptions of steps involved:
+Datanodes and OzoneManagers submits a CSR (certificate signing request) to SCM. 
+SCM verifies identity of DN (Datanode) or OM via Kerberos and generates a certificate. 
+This certificate is used by OM and DN to prove their identities.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: whitespace:end of line

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+Generating audit logs is only half the job, so Ozone also provides AuditParser - a sqllite based command line utility to parse/query audit logs with predefined templates(ex. Top 5 commands) and options for custom query. Once the log file has been loaded to AuditParser, one can simply run a template as shown below:
+ozone auditparser <path to db file> template top5cmds
+
+Similarly, users can also execute custom query using:
+ozone auditparser <path to db file> query "select * from audit where level=='FATAL'"
+
+## Transparent Data Encryption ##
+Ozone TDE borrows heavily from HDFS TDE. So setup process and usage is also very similar. When TDE is enabled an admin can create encryption keys and encryption zones. Any data written to encryption zone is encrypted.
+
+To know more about how to setup a secure Ozone cluster refer to [How to setup secure Ozone cluster]("SetupSecureOzone.md")
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+
+## Kerberos ##
+Configuration for service daemons:
+
+Property|Description
+--------|------------------------------------------------------------
+hdds.scm.kerberos.principal     | The SCM service principal. Ex scm/_HOST@REALM.COM_
+hdds.scm.kerberos.keytab.file   |The keytab file used by SCM daemon to login as its service principal.
+ozone.om.kerberos.principal     |The OzoneManager service principal. Ex om/_HOST@REALM.COM
+ozone.om.kerberos.keytab.file   |The keytab file used by SCM daemon to login as its service principal.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+hdds.scm.http.kerberos.principal|SCM http server service principal. 
+hdds.scm.http.kerberos.keytab   |The keytab file used by SCM http server to login as its service principal.
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+## Tokens ##
+
+## Delegation token ##
+Delegation tokens are enabled by default when security is enabled.
+
+## Block Tokens ##
+hdds.block.token.enabled     | true
+-----------------------------|------
+
+## S3Token ##
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 23]
```diff
@@ -0,0 +1,88 @@
...
+S3 token are enabled by default when security is enabled. 
+To use S3 tokens users need to perform following steps:
+* S3 clients should get the secret access id and user secret from OzoneManager.
+```
+ozone s3 getsecret
+```
+* Setup secret in aws configs:
+```
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 24]
```diff
@@ -0,0 +1,88 @@
...
+aws configure set default.s3.signature_version s3v4
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 25]
```diff
@@ -0,0 +1,88 @@
...
+S3 token are enabled by default when security is enabled. 
+To use S3 tokens users need to perform following steps:
+* S3 clients should get the secret access id and user secret from OzoneManager.
+```
+ozone s3 getsecret
+```
+* Setup secret in aws configs:
+```
+aws configure set default.s3.signature_version s3v4  
+aws configure set aws_access_key_id ${accessId}
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 26]
```diff
@@ -0,0 +1,88 @@
...
+To use S3 tokens users need to perform following steps:
+* S3 clients should get the secret access id and user secret from OzoneManager.
+```
+ozone s3 getsecret
+```
+* Setup secret in aws configs:
+```
+aws configure set default.s3.signature_version s3v4  
+aws configure set aws_access_key_id ${accessId}  
+aws configure set aws_secret_access_key ${secret}
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 27]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+## Certificates ##
+Certificates are used internally inside Ozone. Its enabled be default when security is enabled.
+
+## Authorization ##
+Default access authorizer for Ozone approves every request. It is not suitable for production environments. It is recommended that clients use ranger plugin for Ozone to manage authorizations.
+
+Property|Description
+--------|------------------------------------------------------------
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 28]
```diff
@@ -0,0 +1,88 @@
...
+ozone.acl.enabled         | true
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

File: hadoop-hdds/docs/content/SetupSecureOzone.md [hunk 29]
```diff
@@ -0,0 +1,88 @@
...
+
+## Authorization ##
+Default access authorizer for Ozone approves every request. It is not suitable for production environments. It is recommended that clients use ranger plugin for Ozone to manage authorizations.
+
+Property|Description
+--------|------------------------------------------------------------
+ozone.acl.enabled         | true 
+ozone.acl.authorizer.class| org.apache.ranger.authorization.ozone.authorizer.RangerOzoneAuthorizer   
+
+## TDE ##
```
Comment (hadoop-yetus) [inline hadoop-hdds/docs/content/SetupSecureOzone.md]: whitespace:end of line

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 47 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1132 | trunk passed |
| +1 | mvnsite | 20 | trunk passed |
| +1 | shadedclient | 1839 | branch has no errors when building and testing our client artifacts. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 19 | the patch passed |
| +1 | mvnsite | 16 | the patch passed |
| -1 | whitespace | 0 | The patch has 15 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply |
| +1 | shadedclient | 780 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | asflicense | 25 | The patch does not generate ASF License warnings. |
| | | 2825 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-687/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/687 |
| Optional Tests |  dupname  asflicense  mvnsite  |
| uname | Linux ec7ff6fa3ff6 4.4.0-138-generic #164~14.04.1-Ubuntu SMP Fri Oct 5 08:56:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / be488b6 |
| maven | version: Apache Maven 3.3.9 |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-687/2/artifact/out/whitespace-eol.txt |
| Max. process+thread count | 306 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs U: hadoop-hdds/docs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-687/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 30]
```diff
@@ -0,0 +1,82 @@
...
+Datanodes and OzoneManagers submits a CSR (certificate signing request) to SCM. 
+SCM verifies identity of DN (Datanode) or OM via Kerberos and generates a certificate. 
+This certificate is used by OM and DN to prove their identities. 
+Datanodes use OzoneManager certificate to validate block tokens. This is possible because both of them trust SCM signed certificates. (i.e OzoneManager and Datanodes)
+
+## Authorization ##
+Ozone provides a pluggable API to control authorization of all client related operations. Default implementation allows every request. Clearly it is not meant for production environments. To configure a more fine grained policy one may configure Ranger plugin for Ozone. Since it is a pluggable module clients can also implement their own custom authorization policy and configure it using [ozone.acl.authorizer.class].
+
+## Audit ##
```
[the remark this answers is not in the document]
Comment (dineshchitlangia) [reply hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: +1 for Audit section. Thanks @ajayydv !

[the remark this answers is not in the document]
Comment (ajayydv) [reply hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: thanks @dineshchitlangia

File: hadoop-hdds/docs/content/OzoneSecurityArchitecture.md [hunk 31]
```diff
@@ -0,0 +1,82 @@
...
+Ozone provides ability to audit all read & write operations to OM, SCM and Datanodes. Ozone audit leverages the Marker feature which enables user to selectively audit only READ or WRITE operations by a simple config change without restarting the service(s).
+To enable/disable audit of READ operations, set filter.read.onMatch to NEUTRAL or DENY respectively. Similarly, the audit of WRITE operations can be controlled using filter.write.onMatch.
+
+Generating audit logs is only half the job, so Ozone also provides AuditParser - a sqllite based command line utility to parse/query audit logs with predefined templates(ex. Top 5 commands) and options for custom query. Once the log file has been loaded to AuditParser, one can simply run a template as shown below:
+ozone auditparser <path to db file> template top5cmds
+
+Similarly, users can also execute custom query using:
+ozone auditparser <path to db file> query "select * from audit where level=='FATAL'"
+
+## Transparent Data Encryption ##
```
[the remark this answers is not in the document]
Comment (ajayydv) [reply hadoop-hdds/docs/content/OzoneSecurityArchitecture.md]: updated

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 24 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 975 | trunk passed |
| +1 | mvnsite | 20 | trunk passed |
| +1 | shadedclient | 1584 | branch has no errors when building and testing our client artifacts. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 18 | the patch passed |
| +1 | mvnsite | 15 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 687 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | asflicense | 22 | The patch does not generate ASF License warnings. |
| | | 2430 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-687/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/687 |
| Optional Tests |  dupname  asflicense  mvnsite  |
| uname | Linux 52e1826f9230 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 215ffc7 |
| maven | version: Apache Maven 3.3.9 |
| Max. process+thread count | 443 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs U: hadoop-hdds/docs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-687/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**